packages/gatsby/src/redux/reducers/status.ts

```diff
@@ -42,6 +43,11 @@ export const statusReducer = (
           ),
         },
       }
+    case `CREATE_NODE`:
+      return {
+        ...state,
+        LAST_NODE_COUNTER: action.payload.internal.counter,
+      }
```

Comment:
Should we consider not creating new state object per each created node for perf reasons? (we mutate state in other reducers already for that reason already, so our state is not immutable anyway)
```suggestion
      state.LAST_NODE_COUNTER = action.payload.internal.counter
      return state
```